browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -202,7 +202,7 @@ EngineFlags ShouldBlockRequestOnTaskRunner(
           previous_result.did_match_important);
 
   bool has_valid_rewritten_url = false;
-  if (adblock_result.rewritten_url.has_value &&
+  if (!canonical_url && adblock_result.rewritten_url.has_value &&
```

Comment:
pls add a comment and/or mention issue, so it would be easier to understand why `canonical_url` is here